php-frontend/src/main/java/org/sonar/php/parser/PHPGrammar.java

```diff
@@ -367,6 +368,7 @@ public ClassPropertyDeclarationTree CLASS_CONSTANT_DECLARATION() {
   public ConstantDeclarationTree CONSTANT_DECLARATION() {
     return b.<ConstantDeclarationTree>nonterminal(PHPLexicalGrammar.CONSTANT_DECLARATION).is(
       f.constantDeclaration(
+        b.optional(b.token(PHPKeyword.FINAL)),
```

Comment:
If I understood the RFC correctly, this should only affect class constants. Using `final` on constants that are not class members does not seem possible.